### ISSUE TITLE
Symbol asks for memory all the time when "bind"..

### ISSUE THREAD

**Issue by saicoco** [dmlc/mxnet#5879]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:
ubuntu 15.10
Compiler:
gcc 4.9.x
Package used (Python/R/Scala/Julia):
Python
MXNet version:
0.9.4
Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`):
2ad9a29640cf8677676be67e75dc82cc2f5c1bec
If you are using python package, please provide

Python version and distribution:
2.7
If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.
NO error message
## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.
```python
# coding=utf-8
import mxnet as mx
from collections import namedtuple

def img_sym(convi, conv_w, conv_b, fc_w, fc_b, hidden=256):
    im_filter_list = [3, 3, 3]
    im_num_filters = [64, 128, 256]
    for i, (filter_size, num_filters) in enumerate(zip(im_filter_list, im_num_filters)):
        convi = mx.sym.Convolution(
            data=convi,
            kernel=(filter_size, filter_size),
            num_filter=num_filters,
            weight=conv_w[i],
            bias=conv_b[i]
        )
        convi = mx.sym.Activation(
            data=convi,
            act_type='relu'
        )
        convi = mx.sym.Pooling(
            data=convi,
            kernel=(2, 2),
            pool_type='max'
        )
    
    conv_flatten = mx.sym.Flatten(convi)
    dropout=0.5
    fc = mx.sym.FullyConnected(data=conv_flatten, weight=fc_w, bias=fc_b, num_hidden=hidden)
    fc = mx.sym.Activation(data=fc, act_type='relu')
    fc = mx.sym.Dropout(data=fc, p=dropout)
        
    return fc

def info_sym(info, vocab_size, n_embed, embedding, rnn_params, fc_w, fc_b, hidden):
    embed_layer = mx.sym.Embedding(
        data=info, 
        input_dim=vocab_size, 
        output_dim=n_embed,
        weight=embedding
        )
    seq_mod = mx.rnn.SequentialRNNCell()
    num_lstm_layer = 1
    num_hidden = 256
    num_words = 10
    for i in range(num_lstm_layer):
        seq_mod.add(mx.rnn.LSTMCell(num_hidden=num_hidden, prefix='lstm_{}'.format(i), params=rnn_params[i]))
    outputs, state = seq_mod.unroll(length=num_words, inputs=embed_layer, merge_outputs=False)    
    concat = mx.sym.Concat(*outputs, dim=1)

    ######################################################################
    # dropout layer
    dropout=0.5
    h_drop = mx.sym.Dropout(data=concat, p=dropout)
    fc = mx.sym.FullyConnected(data=h_drop, weight=fc_w, bias=fc_b, num_hidden=hidden)
    fc = mx.sym.Activation(data=fc, act_type='relu')
    fc = mx.sym.Dropout(data=fc, p=dropout)
    return fc

def pairwise_model():
    img = mx.sym.Variable('img')
    info = mx.sym.Variable('info')
    label = mx.sym.Variable('label')

    num_conv = 3
    num_fc = 2
    num_lstm=1
    conv_w = []
    conv_b = []
    fc_w = []
    fc_b = []
    embedding = mx.sym.Variable('weight')
    rnn_params = []
    for i in xrange(num_conv):
        conv_w.append(mx.sym.Variable('conv%d_weight'%(i)))
        conv_b.append(mx.sym.Variable('conv%d_bias'%(i)))
    
    for i in xrange(num_fc):
        fc_w.append(mx.sym.Variable('fc%d_weight'%(i)))
        fc_b.append(mx.sym.Variable('fc%d_bias'%(i)))

    for i in xrange(num_lstm):
        rnn_params.append(mx.rnn.RNNParams(prefix='lstm_{}'.format(i)))

    encode_img = img_sym(img, conv_w, conv_b, fc_w[0], fc_b[0], 256)
    encode_info = info_sym(info, 5590, 256, embedding, rnn_params, fc_w[1], fc_b[1], 256)
    
    margin = 1.0
    distance = mx.sym.sum(mx.sym.square(encode_img - encode_info), axis=1)
    per_loss = label * distance + (1 - label) * mx.sym.square(mx.sym.Activation(margin-distance, act_type='relu'))
    contrastive_loss = mx.sym.MakeLoss(0.5 * mx.sym.mean(per_loss))

    outputs = mx.sym.Group([mx.sym.BlockGrad(distance), contrastive_loss])
    return outputs


if __name__=='__main__':
    sym = pairwise_model()
    exe = sym.simple_bind(img=(10, 3, 224, 224), info=(10, 10), label=(10,), ctx=mx.cpu())

```
## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1.
2.
3.

## What have you tried to solve it?

1. move `SequentialRNNCell` into `pairwise_model` to avoid to ask for memory repeately.
2.put all code into a method, however it doesn't work.
3.check my dataIter works well.

The sym is a toy model for metric learning about image and text, `img_sym` is used to extract image feature, so as `info_sym` does extract text feature.
